File: solutions/cpp/392/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() == 0) {
            return true;
        }

        if (t.size() == 0 || s.size() > t.size()) {
            return false;
        }

        int left = 0;
        for (int right = 0; right < t.size(); right++) {
            if (s[left] == t[right]) {
                left++;
                if (left == s.size()) {
                    return true;
                }
            }
        }

        return false;
    }
};

File: solutions/golang/392/solution.go
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}

	if len(t) == 0 || len(s) > len(t) {
		return false
	}

	left := 0
	for right := 0; right < len(t); right++ {
		if s[left] == t[right] {
			left++
			if left == len(s) {
				return true
			}
		}
	}

	return false
}